ui/app/registration/controllers/createPatientController.js

```diff
@@ -51,9 +51,30 @@ angular.module('bahmni.registration')
                     }).value();
                 };
 
-                _.chain(personAttributeTypes)
+                var isDateType = function (personAttributeType) {
+                    return personAttributeType.format === "org.openmrs.util.AttributableDate";
+                };
+
+                var isDefaultValueToday = function (personAttributeType) {
+                    if (defaults[personAttributeType.name].toUpperCase() === "today".toUpperCase()) {
+                        return true;
+                    }
+                    else {
+                        $scope.patient[personAttributeType.name] = '';
+                        return false;
+                    }
+                };
+
+                var setDefaultTodayDate = function (personAttributeType) {
+                    $scope.patient[personAttributeType.name] = new Date();
+                };
+
+                var defaultsWithAnswers = _.chain(personAttributeTypes)
                     .filter(hasDefaultAnswer)
-                    .each(setDefaultAnswer).filter(isConcept).each(setDefaultConcept).value();
+                    .each(setDefaultAnswer).value();
+
+                _.chain(defaultsWithAnswers).filter(isConcept).each(setDefaultConcept).value();
+                _.chain(defaultsWithAnswers).filter(isDateType).filter(isDefaultValueToday).each(setDefaultTodayDate).value();
```

Comment:
@nehagupta2123, can you use one `filter` instead of two. Just trying to avoid multiple loops